fastlane/lib/fastlane/actions/clear_derived_data.rb

```diff
@@ -17,15 +29,22 @@ def self.description
       end
 
       def self.details
-        "Deletes the Derived Data from '~/Library/Developer/Xcode/DerivedData' or a supplied path"
+        "Deletes the Derived Data from path set on Xcode or a supplied path"
       end
 
       def self.available_options
+        path = begin
+                 xcode_preferences['IDECustomDerivedDataLocation']
+               rescue
+                 nil
+               end
```

Comment:
One last thing 🙃 Can we change this begin/rescue to...
```rb
path = xcode_preferences ? xcode_preferences['IDECustomDerivedDataLocation'] : nil
path ||= "~/Library/Developer/Xcode/DerivedData"
```

Reply:
ofc 🥇